src/parameterizations/vertical/MOM_set_viscosity.F90

```diff
@@ -1202,6 +1086,707 @@ subroutine set_viscous_BBL(u, v, h, tv, visc, G, GV, US, CS, pbv)
+                                                   !! the depth of each interface [nondim]
+  type(unit_scale_type),       intent(in)  :: US   !< A dimensional unit scaling type
+  type(set_visc_CS),           intent(in)  :: CS   !< The control structure returned by a previous
+                                                   !! call to set_visc_init.
+
+  ! Local variables
+  real :: crv              ! crv is the curvature of the bottom depth across a
+                           ! cell, times the cell width squared [Z ~> m].
+  real :: crv_3            ! crv/3 [Z ~> m].
+  real :: slope            ! The absolute value of the bottom depth slope across
+                           ! a cell times the cell width [Z ~> m].
+  ! All of the following "volumes" have units of vertical heights because they are normalized
+  ! by the full horizontal area of a velocity cell.
+  real :: Vol_err          ! The error in the volume with the latest estimate of
+                           ! L, or the error for the interface below [Z ~> m].
+  real :: Vol_quit         ! The volume error below which to quit iterating [Z ~> m].
+  real :: Vol_tol          ! A volume error tolerance [Z ~> m].
+  real :: Vol_open         ! The cell volume above which the face is fully open [Z ~> m].
+  real :: Vol_direct       ! With less than Vol_direct [Z ~> m], there is a direct
+                           ! solution of a cubic equation for L.
+  real :: Vol_err_max      ! The volume error for the upper bound on the correct value for L [Z ~> m]
+  real :: Vol_err_min      ! The volume error for the lower bound on the correct value for L [Z ~> m]
+  real :: Vol_0            ! A deeper volume with known width L0 [Z ~> m].
+  real :: dVol             ! vol - Vol_0 [Z ~> m].
+  real :: dV_dL2           ! The partial derivative of volume with L squared
+                           ! evaluated at L=L0 [Z ~> m].
+  real :: L_direct         ! The value of L above volume Vol_direct [nondim].
+  real :: L_max, L_min     ! Upper and lower bounds on the correct value for L [nondim].
+  real :: L0               ! The value of L above volume Vol_0 [nondim].
+  real :: Iapb, Ibma_2     ! Combinations of crv (a) and slope (b) [Z-1 ~> m-1]
+  real :: C24_crv          ! 24/crv [Z-1 ~> m-1].
+  real :: curv_tol         ! Numerator of curvature cubed, used to estimate
+                           ! accuracy of a single L(:) Newton iteration [Z5 ~> m5]
+  real, parameter :: C1_3 = 1.0/3.0, C1_6 = 1.0/6.0 ! Rational constants [nondim]
+  logical :: use_L0, do_one_L_iter  ! Control flags for L(:) Newton iteration
+  integer :: K, nz, itt, maxitt=20
+
+  nz = GV%ke
+
+  ! Each cell extends from x=-1/2 to 1/2, and has a topography
+  ! given by D(x) = crv*x^2 + slope*x + D_vel - crv/12.
+  crv_3 = (Dp + Dm - 2.0*D_vel) ; crv = 3.0*crv_3
+  if (crv >= 0.0) call MOM_error(FATAL, "find_L_open_convex should only be called with a negative curvature.")
+  slope = Dp - Dm
```

Comment:
As with `find_L_open_concave_*`, the inputs could be reworked so that `Dp`, `Dm`, `D_vel` are replaced with `slope`, `crv`, and `Vol_open`.

Since `crv_3` appears in these expressions, it would make more sense to replace `crv` with `crv_3` as an input.

And as before, the `if (crv >= 0.0)` check is probably unnecessary.